### PR TITLE
bug fixes for trunk sims

### PR DESCRIPTION
**Pull request by sampan501**
Fixes #

Changes proposed in this pull request:

- Bug fix: When increasing the dimensions, we don’t get the same values given the same seed.
- Doc fix: Odd values for n_samples give 1 values less than they should
- Bug fix: Values for trunk are the same for the first group but different for the second group

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: sktree/datasets/hyppo.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ class distribution approaches the first class distribution by a factor of :math:
     Parameters
     ----------
     n_samples : int
-        Number of sample to generate.
+        Number of sample to generate. Must be an even number.
```
Comment (adam2392) [inline sktree/datasets/hyppo.py]:
```suggestion
        Number of sample to generate. Must be an even number, else the total number of samples generated will be ``n_samples - 1``
```

Right?

File: sktree/datasets/hyppo.py [hunk 2]
```diff
@@ -197,31 +197,41 @@ class distribution approaches the first class distribution by a factor of :math:
     )
     # the parameters used for each Gaussian in the mixture for each Marron Wand simulation
     norm_params = MarronWandSims(n_dim=n_informative, cov=cov)(simulation)
+    if seed:
+        rngs = [np.random.default_rng(seed=seed + i) for i in range(n_samples // 2)]
+    else:
+        rngs = [rng for _ in range(n_samples // 2)]
```
Comment (adam2392) [inline sktree/datasets/hyppo.py]: Do we not need to also re-seed here? or is `None` handle that?

File: sktree/datasets/hyppo.py [hunk 3]
```diff
@@ -265,7 +275,7 @@ def make_trunk_mixture_classification(
     Parameters
     ----------
     n_samples : int
-        Number of sample to generate.
+        Number of sample to generate. Must be an even number.
```
Comment (adam2392) [inline sktree/datasets/hyppo.py]: Same comment as above

File: sktree/datasets/hyppo.py [hunk 4]
```diff
@@ -365,22 +375,31 @@ def make_trunk_mixture_classification(
         method = "svd"
 
     mixture_idx = rng.choice(2, n_samples // 2, replace=True, shuffle=True, p=[mix, 1 - mix])  # type: ignore
+    if seed:
+        rngs = [np.random.default_rng(seed=seed + i) for i in range(n_samples // 2)]
+    else:
+        rngs = [rng for _ in range(n_samples // 2)]
```
Comment (adam2392) [inline sktree/datasets/hyppo.py]: Same question as above

**Review by adam2392 — COMMENTED** [review]
Nice! Few minor changes and one question and then LGTM

File: sktree/datasets/hyppo.py (hunk 1 above)
Comment (sampan501) [reply sktree/datasets/hyppo.py]: yes

File: sktree/datasets/hyppo.py (hunk 2 above)
Comment (sampan501) [reply sktree/datasets/hyppo.py]: I think we only care about this when we fix a seed right?

File: sktree/datasets/hyppo.py (hunk 3 above)
Comment (sampan501) [reply sktree/datasets/hyppo.py]: https://github.com/neurodata/scikit-tree/pull/236/commits/02af22dcdde829ee52812b9f1f5f6462201281a8

File: sktree/datasets/hyppo.py (hunk 2 above)
Comment (adam2392) [reply sktree/datasets/hyppo.py]: Oh I see. Because if we don't fix the seed then each run will be different anyways?

**Review by adam2392 — APPROVED** [review]

File: sktree/datasets/hyppo.py (hunk 2 above)
Comment (sampan501) [reply sktree/datasets/hyppo.py]: yep

**Comment by sampan501** [comment]
Found some bugs, I'll squash before I ask for another review

**Comment by sampan501** [comment]
Ok, I believe everything should be correct now. tldr of the changes:

1. Unit tests were testing if the same array is equal since the second array was just `n_samples + 1`. Now, it tests if group 1 elements are the same, group 2 elements are the same, and the noise is the same
2. Was defining a new seed and iterating by an `int`. Switched to the `rng.spawn` method instead
3. Ocassionally, I had to reinitalize a new variable with the same generator `int`. This I learned from validating unit tests.

**Review by PSSF23 — APPROVED** [review]

File: sktree/datasets/hyppo.py [hunk 5]
```diff
@@ -199,36 +199,47 @@ class distribution approaches the first class distribution by a factor of :math:
...
+            rng_children.multivariate_normal(*(norm_params[i]), size=1, method=mvg_sampling_method)
+            for i, rng_children in zip(mixture_idx, rng.spawn(n_samples // 2))
         ),
         dtype=np.dtype((float, n_informative)),
     )
 
     # as the dimensionality of the simulations increasing, we are adding more and
     # more noise to the data using the w parameter
     w_vec = np.array([1.0 / np.sqrt(i) for i in range(1, n_informative + 1)])
+    rng_F = np.random.default_rng(seed=seed).spawn(2)
```
Comment (adam2392) [inline sktree/datasets/hyppo.py:213]: Why not use the existing rng to spawn 2 children here vs re defining a rng for F?

Could be useful to inline comment

**Review by adam2392 — APPROVED** [review]
To confirm the separate rng instances are required to keep noise variables the same even if n informative changes? 

Would be nice to document that just as in line comments for anyone who reads it in the future. 

Otw LGTM!

File: sktree/datasets/hyppo.py (hunk 5 above)
Comment (sampan501) [reply sktree/datasets/hyppo.py:213]: I'll comment. It was from when I was evaluating the unit tests; had the same issue of continue to sample from `rng` leading to failed tests

**Comment by sampan501** [comment]
I'll quickly verify that simulations look correct in scatter plots, add the comment, and then merge

File: sktree/datasets/hyppo.py (hunk 5 above)
Comment (sampan501) [reply sktree/datasets/hyppo.py:213]: https://github.com/neurodata/scikit-tree/pull/236/commits/d522ea9e2e7989763544e012ed6dbbd70ede4492

**Comment by sampan501** [comment]
Ok verified! I'll merge when CIs are green

**Pull request merged**